internal/server/http/dump.go

```diff
@@ -0,0 +1,285 @@
+package http
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+	"unicode/utf8"
+
+	"github.com/gorilla/handlers"
+)
+
+var binaryContentTypes = []string{
+	"application/octet-stream",
+	"image/",
+	"audio/",
+	"video/",
+	"application/pdf",
+}
+
+// RequestData struct to hold request data
+type RequestData struct {
+	Method string      `json:"method"`
+	Host   string      `json:"host"`
+	URL    string      `json:"url"`
+	Header http.Header `json:"header"`
+	Body   string      `json:"body"`
+}
+
+func getRequestData(r *http.Request, body string) *RequestData {
+	return &RequestData{
+		Method: r.Method,
+		Host:   r.Host,
+		URL:    r.URL.String(),
+		Header: r.Header,
+		Body:   body,
+	}
+}
+
+// Copied from the gorilla/handlers package
```

Comment:
For all those contents copied over from `gorilla/handlers`, I'd love to see them:
- Isolated as much as possible (ideally in a different package, or at least in a different file/s).
- Accompanied with the reference they were copied from; like the Git hash, so we can track future changes easily.
- Accompanied with the corresponding license, as I think `gorilla/handlers` is licensed with BSD-3 and it requires to retain copyright.
- Unit tests, if any available in the original codebase.

Reply:
Or even better, have you considered re-using `LoggingHandler` instead? 🤔  What prevents you from doing so?

Reply:
So I dropped all the gorilla code and just went with a json logger for now. We can add other formats later when/if wanted. This keeps us from having to deal with multiple license fun and we don't have copy pasted code from another repo. We just reuse the gorilla CustomLoggingHandler to log/dump requests when needed.